plugins/parcelport/mpi/CMakeLists.txt

```diff
@@ -56,11 +57,11 @@ if(HPX_WITH_PARCELPORT_MPI)
     endif()
 
     set(_mpi_libraries)
-    if(MPI_C_LIBRARIES)
-      set(_mpi_libraries ${_mpi_libraries} ${MPI_C_LIBRARIES})
-    elseif(MPI_C_LIB_NAMES)
+    if(MPI_CXX_LIBRARIES)
+      set(_mpi_libraries ${_mpi_libraries} ${MPI_CXX_LIBRARIES})
+    elseif(MPI_CXX_LIB_NAMES)
       # cmake V3.10.2 has changed the way MPI libraries are being discovered
-      foreach(lib ${MPI_C_LIB_NAMES})
```

Comment:
Are the C libraries always included in the CXX library list? What if we encounter a MPI library without CXX bindings at all?

Reply:
Yes, the C libraries are always included. CXX is the C++-compatible C interface. The C++ interface, if it is available, is either always included (CMake < 3.10) or needs to be explicitly requested via the MPICXX component (CMake >= 3.10) as far as CMake is concerned.

The documentation for the MPI find module (https://cmake.org/cmake/help/v3.10/module/FindMPI.html) clearly states this from version 3.10 onward. Unfortunately, the documentation for older versions is not as explicit. But judging by the actual code, the CXX variants of all variables and targets always include the information from the C versions.

Depending on the MPI implementation, the C++ interface might still be included even if only the C interface has been requested. Its existence is just not guaranteed.